lib/async.array.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+const { promisify } = require('util');
+
+const timeout = promisify(resolve => setTimeout(resolve, 0));
+
+class AsyncArray extends Array {
+  constructor(...args) {
+    super(...args);
+
+    this.min = 5;
+    this.ratio = 1.5;
+  }
+
+  set percent(value) {
+    if (value < 1 && value >= 0) {
+      this.ratio = value / (1 - value);
+    }
+  }
+
+  [Symbol.asyncIterator]() {
+    let sum = 0;
+    let count = 0;
+    let begin = Date.now();
+    let iterMax = this.min;
+
+    const getItem = () => {
+      const item = { value: this[count], done: count === this.length };
```

Comment:
```suggestion
      const item = { value: this[count], done: count >= this.length };
```
AsyncIterator interface defines that any consecutive calls to `next` after consuming an iterrators should return:
```js
{ value: undefined, done: false }
```